cluster/manifests/z-karpenter/deployment.yaml

```diff
@@ -0,0 +1,125 @@
+{{ if eq .Cluster.ConfigItems.karpenter_pools_enabled "true"}}
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: karpenter
+  namespace: kube-system
+  labels:
+    application: kubernetes
+    component: karpenter
+spec:
+  replicas: 2
+  strategy:
+    rollingUpdate:
+      maxUnavailable: 1
+  selector:
+    matchLabels:
+      deployment: karpenter
+  template:
+    metadata:
+      labels:
+        deployment: karpenter
+        application: kubernetes
+        component: karpenter
+      annotations:
+        logging/destination: "{{.Cluster.ConfigItems.log_destination_infra}}"
+    spec:
+      dnsConfig:
+        options:
+          - name: ndots
+            value: "1"
+      serviceAccountName: karpenter
+      securityContext:
+        fsGroup: 1000
+      priorityClassName: "{{ .Cluster.ConfigItems.system_priority_class }}"
+      tolerations:
+      - key: node.kubernetes.io/role
+        value: master
+        effect: NoSchedule
+      containers:
+        - name: controller
+          image: "public.ecr.aws/karpenter/controller:v0.16.1@sha256:ed1f97bc2cd67e63b45546955cf763890182f1f97586fca5e6cbd25f175b9c56"
```

Comment:
Now there is v0.18.0 :sweat_smile: https://github.com/aws/karpenter/releases/tag/v0.18.0

Reply:
nit: we also need to have a copy of this in our registry

Reply:
yes, I noticed. but then decided to merge this one and update later. it is getting bigger and bigger 😆

Reply:
Fixed in https://github.com/zalando-incubator/kubernetes-on-aws/pull/5446